### PR TITLE
Intern Guide: Preparing the final application

### PR DESCRIPTION
**Pull request by sgibson91**
Add guide for preparing final application

Covers ideas discussed in our [retrospective](https://hackmd.io/@sgibson91/S1RyCF5ro), specifically around community questions and using the timeline section of the application to our advantage

relates to #45 

[Link to rendered docs](https://jupyterhub-outreachy--142.org.readthedocs.build/en/142/interns/prepare-application.html)

### CODE REVIEW

File: docs/source/interns/prepare-application.md [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+(application-prep:put-together:comm-qs)=
+
+### Community-specific questions
+
+Please answer these questions in the relevant section of your application:
+
+1. **In your own words, what problem is the project attempting to solve?**
+   (_50 words max_)
+2. **In your own words, what is the main goal of the project?** (_50 words max_)
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:63]: Are these the best questions we could ask? What could we ask instead?

File: docs/source/interns/prepare-application.md [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+need to be solved in order to be successful.
+
+Your outline should include:
+
+- A clear plan of action. What are you going to actually do?
+- Milestones that are achievable and realistic
+- Things happen! We can't foresee every problem! Applications that say "I will
+  do A and B. If there is time, I will do C." are encouraged
+
+Here are a couple of successful applications to help you
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:90]:
```suggestion
Here are a couple of successful applications to help you:
```

**Comment by sgibson91** [comment]
@yoyehudi if you have thoughts, I would love to hear them! 🙏🏻

File: docs/source/interns/prepare-application.md [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+- A clear plan of action. What are you going to actually do?
+- Milestones that are achievable and realistic
+- Things happen! We can't foresee every problem! Applications that say "I will
+  do A and B. If there is time, I will do C." are encouraged
+
+Here are a couple of successful applications to help you:
+
+- [2018 GSoC applicants to InterMine](https://github.com/nupurgunwant/GSoC-Proposal)
+  - This is a little bit different from the Outreachy application, but should
+    help you generate some ideas
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:94]:
```suggestion
- [2018 GSoC applicants to InterMine](https://github.com/nupurgunwant/GSoC-Proposal)
  - This is a little bit different from the Outreachy application, but should
    help you generate some ideas
- [2022 Outreachy applicant to OLS](https://docs.google.com/document/d/1CBGxozCCEdysdO-WXkmJtXchOZ3QiTUM-9jMtxqNKSg/edit?usp=sharing)
```

File: docs/source/interns/prepare-application.md [hunk 4]
```diff
@@ -0,0 +1,128 @@
...
+need to be solved in order to be successful.
+
+Your outline should include:
+
+- A clear plan of action. What are you going to actually do?
+- Milestones that are achievable and realistic
+- Things happen! We can't foresee every problem! Applications that say "I will
+  do A and B. If there is time, I will do C." are encouraged
+
+Here are a couple of successful applications to help you:
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:90]:
```suggestion
Here are a couple of successful applications to help you (Thank you to the applicants
for sharing!):
```

**Comment by sgibson91** [comment]
@alwasega @sheila-nk @bl-aire I wondered if you three could review this documentation and let us know if having this information would've been useful during your application? Thank you!

File: docs/source/interns/prepare-application.md [hunk 5]
```diff
@@ -0,0 +1,130 @@
...
+    [InterMine](http://intermine.org/internships/guidance/students-applying-outreachy/).
+    Thank you for developing such awesome resources!
+
+This section contains information on how to prepare your final application
+_specifically for the JupyterHub community_.
+
+```{seealso}
+Outreachy provides their own, more general
+[guide on how to complete your Final Application](https://www.outreachy.org/docs/applicant/#final-application).
+Please make to read through this as well.
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:17]:
```suggestion
Please make sure to read through this as well.
```

File: docs/source/interns/prepare-application.md (hunk 1 above)
Comment (alwasega) [reply docs/source/interns/prepare-application.md:63]: I like the questions. Can you also consider one related to what success looks like for the intern within the specific project? I believe such a question would help in evaluating the applicant's understanding of the project's problem and deliverable(s).

**Comment by alwasega** [comment]
> @alwasega @Sheila-nk @bl-aire I wondered if you three could review this documentation and let us know if having this information would've been useful during your application? Thank you!

The information would have been very useful. I, especially, remember having trouble with my project outline and having the samples as included here would have really helped.

I noted some unclear parts in the documentation though: 

> Start developing your final application only once you have experience iterating on your pull requests to get them ready to merge.

As an applicant, I will be wondering what this really means. Can it also be made clear that PRs do not have to be merged before one can work on their final application. 

> We highly recommend sharing a rough draft of your application at least one week before the deadline. 

Please include the channels through which one can share the drafts.

**Comment by Sheila-nk** [comment]
> @alwasega @Sheila-nk @bl-aire I wondered if you three could review this documentation and let us know if having this information would've been useful during your application? Thank you!

👏 This is really great and useful work. It is quite comprehensive and will be handy during the final application process. I like that you added examples of successful applications. I also like that the applicants are encouraged to share their final applications with the community for feedback. 

Apart from the part that @alwasega mentioned about "experience iterating on your pull requests", no notes from me. ❤️

File: docs/source/interns/prepare-application.md [hunk 6]
```diff
@@ -0,0 +1,130 @@
...
+    for example, Python
+  - Your **prior contributions to open source projects** (including pull requests,
+    bug reports, and so on) with links, if applicable (don't worry if this is
+    your first time contributing!)
+  - Any other **materials which will help us evaluate how you work**, such as
+    links to personal or school projects, along with brief descriptions
+- Your **contributions to JupyterHub**, including pull requests, bug reports, and
+  helping others in the community (with links to all materials)
+- Your answers to the [](application-prep:put-together:comm-qs)
+- A [](application-prep:put-together:proj-outline)
```
Comment (minrk) [inline docs/source/interns/prepare-application.md:53]: I didn't know you could get auto link text this way!

File: docs/source/interns/prepare-application.md (hunk 1 above)
Comment (minrk) [reply docs/source/interns/prepare-application.md:63]: I agree that a question that gets at trying to describe concretely what the end of a successful internship would be helpful.

**Review by minrk — APPROVED** [review]
This seems like a great resource, thanks @sgibson91!

**Comment by sgibson91** [comment]
> > We highly recommend sharing a rough draft of your application at least one week before the deadline.
> 
> Please include the channels through which one can share the drafts.

@alwasega We are working on getting the channels set up in this issue, but it hasn't had a lot of traction yet 😕 https://github.com/jupyterhub/team-compass/issues/584

**Comment by bl-aire** [comment]
> @alwasega @Sheila-nk @bl-aire I wondered if you three could review this documentation and let us know if having this information would've been useful during your application? Thank you!

This would have been really helpful. 
I am sure it will help future applicants because it will make the application process feel less overwhelming. Steps to take and even samples of successful applications are provided which they can use as templates to tell their unique stories.

**Comment by alwasega** [comment]
> @alwasega We are working on getting the channels set up in this issue, but it hasn't had a lot of traction yet 😕 https://github.com/jupyterhub/team-compass/issues/584

Okay :) . I would like to be a part of the group that assists with answering questions during the contribution phase. Will check the issue in more detail and give my thoughts.

File: docs/source/interns/prepare-application.md [hunk 7]
```diff
@@ -0,0 +1,149 @@
...
+And sometimes you may be asked to make further changes once the first set of
+feedback has been addressed. This is what we mean by "iterating on your pull
+requests" - it's referring to this feedback cycle.
+
+This feedback cycle is normal behaviour is most open source communities and,
+generally speaking, maintainers try to keep the cycle short. You may be asked
+to open new issues or new pull requests to track pieces of feedback that the
+team would like to include in the future, but don't think it is necessary to
+include in your current pull request.
+```
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:41]: @alwasega Does this new section clarify your question around what it means to iterate on a PR?

File: docs/source/interns/prepare-application.md (hunk 1 above)
Comment (sgibson91) [reply docs/source/interns/prepare-application.md:80]: I have added a new question to this section, what do you both think?

Comment (alwasega) [reply docs/source/interns/prepare-application.md:80]: The added question looks great :)

File: docs/source/interns/prepare-application.md (hunk 7 above)
Comment (alwasega) [reply docs/source/interns/prepare-application.md:41]: Yes, it does. I also like that you have added that the PRs do not have to be merged for one to work on their final applications.

File: docs/source/interns/prepare-application.md [hunk 8]
```diff
@@ -0,0 +1,149 @@
...
+
+Your first priority during the contribution period should be figuring out how
+to become an effective contributor. Start developing your final application only
+once you have experience iterating on your pull requests to get them ready to
+merge. This way, you'll have a much better idea of how much you can accomplish
+and how you may wish to tackle your chosen project. **It is ok if your pull
+requests have not been merged when you begin working on your final application,
+or if they are not merged before the application deadline.**
+
+```{note} What does iterating on your pull request mean?
```
Comment (sgibson91) [inline docs/source/interns/prepare-application.md:28]:
```suggestion
```{admonition} What does iterating on your pull request mean?
```

**Comment by sgibson91** [comment]
Thanks everyone!

**Pull request merged**